### ISSUE TITLE
[2013-03-28] More problems with shortcodeparser

### ISSUE THREAD

**Issue by silverstripe-issues** [silverstripe/silverstripe-cms#670]
created by: @Josua2012 (josua)
created at: 2013-03-28
original ticket: http://open.silverstripe.org/ticket/8352

---

$text = '[SendaBackground type='standard' name='msg1' location='left']Esto es una prueba de background2[/SendaBackground]';
$ss = ShortcodeParser::get_active()->parse($text);
## Fatal error: Call to a member function insertBefore() on a non-object in C:\wamp\www\framework\parsers\ShortcodeParser.php on line 424

$text = '[SendaBackground type='standard' name='msg1' location='right']Esto es una prueba de background2[/SendaBackground]';
$ss = ShortcodeParser::get_active()->parse($text);

Fatal error: Call to a member function insertBefore() on a non-object in C:\wamp\www\framework\parsers\ShortcodeParser.php on line 424

---

$text = '[SendaBackground type='standard' name='msg1' location='center']Esto es una prueba de background2[/SendaBackground]';
$ss = ShortcodeParser::get_active()->parse($text);

Fatal error: Call to a member function appendChild() on a non-object in C:\wamp\www\framework\parsers\ShortcodeParser.php on line 121

---

In short, the shortcodes parser is terrible. It is full of bugs.
It is useless to do anything decent with it.
I think it needs a complete overhaul. :(

Regards,
Jose A.
